doc/design/GeneratedClientDesign.md

```diff
@@ -0,0 +1,133 @@
+
+### Type mappings from Cadl to .NET
+
+The following table shows a mapping from Cadl built-in types to the corresponding types used in .NET models.
+
+Cadl Type | .NET Type | OpenAPI Type | GitHub Issue | Notes
+------------------- | -------- | -- | -- | -------------
+string | string | string | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) | [PrimitivePropertyModel.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.cs) [PrimitivePropertyModel.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.Serialization.cs)
+bytes | bytes[] | type: string, format: byte | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+int32  | int | type: integer, format: int32 | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+int64  | long | type: integer, format: int64 | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+safeint  | long | n/a | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+float32  | float | type: number, format: float | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+float64  | double | type: number, format: double | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+zonedDateTime  | DateTimeOffset | type: string, format: date-time | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) | Serialized differently based on body/header
+duration  | TimeSpan | type: string, format: duration | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+boolean  | boolean | boolean | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+<!-- TBD | TBD | type: string, format: binary | [autorest.csharp #2500](https://github.com/Azure/autorest.csharp/issues/2500)| -->
+
+### Model Shape
+
+Model APIs will differ depending on whether the model is an input model, an output model, or a round-trip model (both input and output).
+
+The following describes the model shapes for different cases of properties.
+
+ Item |  Input | Output | Round-Trip | GitHub Issue
+-- | -------| ------------ | -------- | -- |
+Constructor Accessibility | public | internal | public |
+Required Property | get-only | get-only | get/set | [autorest.csharp #2463](https://github.com/Azure/autorest.csharp/issues/2463)
+Optional Property | get/set | get-only | get/set | [autorest.csharp #2339](https://github.com/Azure/autorest.csharp/issues/2339)
+Collection Property | `IList<T>` get-only | `IReadOnlyList<T>` get-only | `IList<T>` get-only | [autorest.csharp #2471](https://github.com/Azure/autorest.csharp/issues/2471)
```

Comment:
Proposal - Input models have get/set IList

Negative of settable collection properties
- may be a collection that's not a known type, e.g. could not retrieve items
- easier to modify the collection from a different thread
- could lose observability

We should decide this as a team, should we change from what we've done in the past?
- @annelo-msft to file an issue to track this in the GA timeframe